marklogic-data-hub/src/main/resources/scaffolding/custom-module/sjs/main-custom.sjs

```diff
@@ -1,125 +1,64 @@
-  if (!fn.docAvailable(id)) {
-    datahub.debug.log({message: 'The document with the uri: ' + id + ' could not be found.', type: 'error'});
-    throw Error('The document with the uri: ' + id + ' could not be found.')
-  }
-  */
-
-  //grab the 'doc' from the content value space
-  let doc = content.value;
-
-  // let's just grab the root of the document if its a Document and not a type of Node (ObjectNode or XMLNode)
-  if (doc && (doc instanceof Document || doc instanceof XMLDocument)) {
-    doc = fn.head(doc.root);
-  }
-
-  //get our instance, default shape of envelope is envelope/instance, else it'll return an empty object/array
-  //If 'doc' is an xml document,  use flowUtils.getInstance(doc)
-  let instance = flowUtils.getInstanceAsObject(doc) || {};
-
-  // get triples, return null if empty or cannot be found
-  //If 'doc' is an xml document,  use flowUtils.getTriples(doc)
-  let triples = flowUtils.getTriplesAsObject(doc) || [];
-
-  //gets headers, return null if cannot be found
-  //If 'doc' is an xml document,  use flowUtils.getHeaders(doc)
-  let headers = flowUtils.getHeadersAsObject(doc) || {};
-
-  //If your instance is xml and you want to set attachments, you will first want to convert it to json.
-  //instance = flowUtils.xmlToJson(instance);
-
-  //If you want to set attachments, uncomment here
-  // instance['$attachments'] = doc;
-
-  //for ES models, you will want to specify entity/version if they are not already part of your instance
-  //instance['$type'] = 'myEntity';
-  //instance['$version'] = '0.0.1'
-
-
-  //insert code to manipulate the instance, triples, headers, uri, context metadata, etc.
-
-
-  //form our envelope here now, specifying our output format
-  let envelope = flowUtils.makeEnvelope(instance, headers, triples, outputFormat);
-
-  //create our return content object, we have a handy helper function for creating a json scaffolding, but you
-  //can also do a node-based one by using nodebuilder, especially if you're dealing with xml!
-  let newContent = flowUtils.createContentAsObject();
-
-  //assign our envelope value
-  newContent.value = envelope;
-
-  //assign the uri we want, in this case the same
-  newContent.uri = id;
-
-  //assign the context we want
-  newContent.context = context;
-
-  //now let's return out our content to be written, it can be any combination of
-  return newContent;
+  const inputDocument = content.value;
+
+  // DHF recommends wrapping documents in an envelope, particularly for curated documents based on an entity model.
+  // If your input document is already an envelope, you can typically just call toObject() it and modify what you need.
+  // Otherwise, the below code is a starting point for constructing the 3 parts of an envelope.
+  // If your input document is XML and you need to modify it, it is recommended to generate an XQuery custom step instead.´
+  const instance = inputDocument.toObject();
+  const headers = {};
+  const triples = [];
```

Comment:
Let's include an example here of how to use .xpath() to access data in an XML document, as that is a common use case and typically not well-known by a new ML developer.